studiopress/pattern-manager
# Check for version control, display notice

This PR adds a dismissible notice in `App` if a `.git` folder is not found in the currently active theme directory:

<img width="1111" alt="Screenshot 2023-05-01 at 11 43 32 AM" src="https://user-images.githubusercontent.com/108079556/235490174-b9bbc59d-1289-4e47-9a67-7f9df7d94a54.png">

---

### Initial spike questions:
1. ~Should the notice be dismissible? If so, should the dismissed state persist for a given theme?~ -> **both items done**
2. ~Should the notice should also be shown in `Editor`?~ -> **notice only shown in `App`**
3. ~Should `.svn` or `.hg` be included in the check?~ -> **not for this version**
4. ~What helper guide should the helper link point to?~ -> **deferred**

#### Note about helper link deferral
While it would be a good idea to add a link to a basic guide for using git with a WordPress theme, finding a place to host that article will hold up the merging of this PR. A helper can link can be added in the future.

---

### To-do
- [x] Make the dismissed notice state persist for a given theme
- [x] Translate `VersionControlNotice` strings
- [x] Add basic tests for `get-version-control` wp-module

---

### How to test

1. Checkout the branch
2. Activate a theme with no `.git` folder in the directory
3. The notice should display
4. Dismiss the notice and refresh the page
6. No notice should display (the dismissed state persisted)
